pkg/query-service/app/http_handler.go

```diff
@@ -2192,6 +2193,130 @@ func (aH *APIHandler) WriteJSON(w http.ResponseWriter, r *http.Request, response
 	w.Write(resp)
 }
 
+// Preferences
+func (ah *APIHandler) RegisterPreferenceRoutes(router *mux.Router, am *AuthMiddleware) {
+	subRouter := router.PathPrefix("/api/v1").Subrouter()
+
+	// user actions
+	subRouter.HandleFunc("/user/preferences", am.ViewAccess(ah.getAllUserPreferences)).Methods(http.MethodGet)
+
+	subRouter.HandleFunc("/user/preferences/{preferenceId}", am.ViewAccess(ah.getUserPreference)).Methods(http.MethodGet)
+
+	subRouter.HandleFunc("/user/preferences/{preferenceId}", am.ViewAccess(ah.updateUserPreference)).Methods(http.MethodPost)
+
+	// org actions
+	subRouter.HandleFunc("/org/preferences", am.AdminAccess(ah.getAllOrgPreferences)).Methods(http.MethodGet)
+
+	subRouter.HandleFunc("/org/preferences/{preferenceId}", am.AdminAccess(ah.getOrgPreference)).Methods(http.MethodGet)
+
+	subRouter.HandleFunc("/org/preferences/{preferenceId}", am.AdminAccess(ah.updateOrgPreference)).Methods(http.MethodPost)
+}
+
+func (ah *APIHandler) getUserPreference(
+	w http.ResponseWriter, r *http.Request,
+) {
+	preferenceId := mux.Vars(r)["preferenceId"]
+	orgId := r.URL.Query().Get("orgId")
+	preference, apiErr := preferences.GetUserPreference(
+		r.Context(), preferenceId, orgId,
+	)
+	if apiErr != nil {
+		RespondError(w, apiErr, nil)
+		return
+	}
+
+	ah.Respond(w, preference)
+}
+
+func (ah *APIHandler) updateUserPreference(
+	w http.ResponseWriter, r *http.Request,
+) {
+	preferenceId := mux.Vars(r)["preferenceId"]
+	req := preferences.UpdatePreference{}
+
+	err := json.NewDecoder(r.Body).Decode(&req)
+
+	if err != nil {
+		RespondError(w, model.BadRequest(err), nil)
+		return
+	}
+	preference, apiErr := preferences.UpdateUserPreference(r.Context(), preferenceId, req.PreferenceValue)
+	if apiErr != nil {
+		RespondError(w, apiErr, nil)
+		return
+	}
+
+	ah.Respond(w, preference)
+}
+
+func (ah *APIHandler) getAllUserPreferences(
+	w http.ResponseWriter, r *http.Request,
+) {
+	orgId := r.URL.Query().Get("orgId")
+	preference, apiErr := preferences.GetAllUserPreferences(
+		r.Context(), orgId,
+	)
+	if apiErr != nil {
+		RespondError(w, apiErr, nil)
+		return
+	}
+
+	ah.Respond(w, preference)
+}
+
+func (ah *APIHandler) getOrgPreference(
+	w http.ResponseWriter, r *http.Request,
+) {
+	preferenceId := mux.Vars(r)["preferenceId"]
+	orgId := r.URL.Query().Get("orgId")
```

Comment:
So does that mean the url is /org/preferences?orgId=<id>?

If yes, again it makes more idomatic sense to move it to /org/{orgId}/preferences!

Reply:
no it won't be the same. replied above